### PR TITLE
Adds GH action CI with path filters and moves CircleCI to nightly only

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
This repository contains components which are not related to e2e CI test we run including 
- Kubernetes directory which contains helm deployment
  - sub-directories
- README.md and other files

We trigger circleCI workflow on each commit irrespective of it's relevance in CI. This is resource heavy and eats up our circleCI quota. 

This PR,
- adds GitHub actions based CI which will run whenever we commit in `\docker` or \.github` directory. 
- moves circleCI to nightly only so it will run ones a day like it used to earlier. 

### Testing
- did test commit by making changes in README to see if it triggers github action.

### Checklist:
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Any dependent changes have been merged and published in downstream modules

### Documentation
- I have tried path filters for circleCI using this: https://github.com/chamindasliit/embassadar/blob/c4733bb8fe27525f38fcb31e95e9e4527b5beb9a/.circleci/config.yml.d/util.yml#L180-L202 (suggested by @adriancole ) but it is giving some weird error which consumed enough of my time. 
- path filters with GH actions are lot simpler: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

So, I chose to go with GH action here.

### CODE REVIEW

File: .github/workflows/docker-tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          docker_buildx: false
+      - name: Cache docker
+        uses: actions/cache@v1
+        with:
+          path: ~/.docker
+          key: ${{ runner.os }}-docker-${{ hashFiles('**/Dockerfile') }}
+          restore-keys: ${{ runner.os }}-docker
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install JDK
```
Comment (kotharironak) [inline .github/workflows/docker-tests.yml:36]: Do we need these steps of installing JDK?

Comment (JBAhire) [reply .github/workflows/docker-tests.yml]: removed

**Review by jcchavezs — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]

**Comment by jcchavezs** [comment]
Thanks a lot @JBAhire!

File: .github/workflows/docker-tests.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: e2e test
+on:
+  push:
+    paths:
+      - 'docker/**'
+      - '.github/**'
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:6]: good memory :) easy to forget to add the file itself

File: .github/workflows/docker-tests.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  build-on-linux:
+    strategy:
+      matrix:
+        docker-version: [19.03]
+    runs-on: ubuntu-latest
+    name: Build and run Docker images
+    steps:
+      - name: Update Packages
+        run: sudo apt-get update -yqq --fix-missing
+      - name: Install Docker
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:23]: fwiw I would remove this step, and the matrix as 19.03 is installed by default and installing takes time

**Pull request merged**